packages/app-project/components/ClassifyPage/components/ProjectStatistics/components/CompletionBar/CompletionBar.js

```diff
@@ -0,0 +1,89 @@
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
+import * as d3 from 'd3'
+import theme from '@zooniverse/grommet-theme'
+import styled from 'styled-components'
+
+import addBackgroundLayer from './d3/addBackgroundLayer'
+import createBar from './d3/createBar'
+import createLabel from './d3/createLabel'
+
+const SVG = styled.svg`
+  display: block;
+`
+
+class CompletionBar extends Component {
+  constructor () {
+    super()
+    this.svgRef = React.createRef()
+    this.d3svg = null
+  }
+
+  componentDidMount () {
+    this.initChart()
+    this.drawChart()
+  }
+
+  initChart () {
+    this.d3svg = d3.select(this.svgRef.current)
+
+    this.d3svg
+      .selectAll('.bar')
+      .data([this.props.completeness])
+      .enter()
+      .append('g')
+      .attr('class', 'bar')
+      .call(createBar, theme.global.colors.teal)
+      .call(createLabel, theme.global.colors.teal)
+  }
+
+  drawChart () {
+    const bar = this.d3svg.selectAll('.bar')
+
+    // TODO: see if this looks better when using a scale transform instead of
+    // manipulating the width.
+    bar.selectAll('rect')
+      .transition()
+      .duration(1000)
+      .attr('width', d => `${d * 100}%`)
+
+    bar.selectAll('text')
+      .transition()
+      .duration(1000)
+      .tween('text', (d, i, elements) => {
+        // `this` _should_ be bound to the element, but in React is already
+        // bound to the component instance, so we define it here as `node`.
```

Comment:
Minor: is the reason that `this` is bound to the React component, instead of the element, due to how we're using `() => {}` instead of `function () {}` ?

Reply:
Hah - yes, that's exactly it, I'm going to fix this up. Good catch.